script/script-runner/package.json

```diff
@@ -0,0 +1,7 @@
+{
+  "name": "atom-build-scripts-runner",
+  "description": "Atom build scripts runner",
+  "dependencies": {
+    "threads": "^1.6.3"
+  }
+}
```

Comment:
Why is this a separate package, instead of a dependency in `scripts`? (I know there may be a reason, I just don't see it written anywhere).

Reply:
The script-runner needs to be installed before running the scripts and so it should be in a separate folder.

> npm install

You should never use `npm install` to install the script dependencies. That is why we have `bootstrap` after all. Our custom bootstrap script installs the dependencies in order and builds them against the Electron version. Using `npm install` bypasses all of these which should be avoided.

Reply:
> The script-runner needs to be installed before running the scripts

Then I would say this is all the more reason not to use it in the bootstrap. Unless you want to make a `bootstrap-bootstrap` task?

Reply:
I don't understand your reason here. This is similar to using `gulp, grunt, etc` to run the scripts. The difference is that `threads` is multi-threaded and does not require writing the scripts in a certain way.

The script runner is not specific to bootstrap only. It is used to [build](https://github.com/atom-ide-community/atom/pull/155) or any other script.

Reply:
Yeah, the bootstrap script does currently use things that require installed modules. I dislike that conceptually (hence the comment), but adding more is not a deal breaker on this PR.

I would still appreciate a comment somewhere in the files or on the introductory PR comment to explain it in a more discoverable place.  

> The script runner is not specific to bootstrap only. It is used to build or any other script.

I was requested to review this repo, not the community one. I understand you intend to use it here with the `build` script too, but that's not been merged into this repo yet so is irrelevant as far as I am concerned. That PR can introduce the script runners itself if they are not already introduced by some earlier PR.

Reply:
With what I have illustrated in [this comment](https://github.com/atom/atom/pull/21308#discussion_r494086785), I am moving towards clearing up Atom's scripts/lib. For that, I need to use a folder that installs the script-runner (lib) files.

There is no reason to write the code manually and maintain 10 packages for that when the same functionality can be packed somewhere else, and you just bring and use it in a clean and easy way.

If you are still considering your [logging PR](https://github.com/atom/atom/pull/21288), you should do the same thing. That task file should be a seprate package.